src/charts/utils/loadingGraph.ts

```diff
@@ -0,0 +1,47 @@
+import type { TimeRange, Timeseries } from "fiberplane-charts";
+import type { Result } from "../../providerRuntime/types";
+import { getNonce } from "../../utils";
+import { vscode } from "../chart";
+
+const requests: Record<string, (result: Result<Timeseries[], String>) => void> =
```

Comment:
i'm not sure how often items are changed in `requests`, but it might be beneficial performance-wise to use Map here instead